test/e2e/rules_test.go

```diff
@@ -0,0 +1,170 @@
+// +build integration
+
+package e2e
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/efficientgo/e2e"
+	"github.com/efficientgo/tools/core/pkg/testutil"
+)
+
+func TestRulesAPI(t *testing.T) {
+	t.Parallel()
+
+	e, err := e2e.NewDockerEnvironment(envRulesAPIName)
+	testutil.Ok(t, err)
+	t.Cleanup(e.Close)
+
+	prepareConfigsAndCerts(t, rules, e)
+	_, token, rateLimiterAddr := startBaseServices(t, e, rules)
+	rulesEndpoint := startServicesForRules(t, e)
+
+	api, err := newObservatoriumAPIService(
+		e,
+		withRulesEndpoint("http://"+rulesEndpoint),
+		withRateLimiter(rateLimiterAddr),
+	)
+	testutil.Ok(t, err)
+	testutil.Ok(t, e2e.StartAndWaitReady(api))
+
+	rulesEndpointURL := "https://" + api.Endpoint("https") + "/api/metrics/v1/test-oidc/api/v1/rules/raw"
+	tr := &http.Transport{
+		TLSClientConfig: getTLSClientConfig(t, e),
+	}
+
+	client := &http.Client{
+		Transport: &tokenRoundTripper{rt: tr, token: token},
+	}
+
+	t.Run("get-put-recording-rules", func(t *testing.T) {
```

Comment:
`get-put-recording-rules` this is kind of confusing to me as to what the test is aiming to do?

Reply:
I renamed to `read-write-recording-rules` (also to the further ones) - hopefully now is clearer?
It basically tests the reading (`GET`) / writing (`PUT`) of rules via the rules API defined in observatorium/api going through `rules-objectstore` https://github.com/observatorium/api/pull/198/files#diff-368252fd7020798243fc9b3793b03207f23440bf25424123e38b77eabf42c3d3R28